packages/nouns-contracts/test/foundry/NounsDAOLogicV3.t.sol

```diff
@@ -0,0 +1,383 @@
+        bytes[] memory calldatas = new bytes[](11);
+        for (uint256 i = 0; i < 11; ++i) {
+            targets[i] = makeAddr('target');
+            values[i] = i;
+            signatures[i] = '';
+            calldatas[i] = '';
+        }
+
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.TooManyActions.selector));
+        vm.prank(proposer);
+        dao.updateProposal(proposalId, targets, values, signatures, calldatas, '');
+    }
+
+    function test_givenTxsWithArityMismatch_reverts() public {
+        address[] memory targets = new address[](1);
+        targets[0] = makeAddr('target');
+        uint256[] memory values = new uint256[](0);
+        string[] memory signatures = new string[](0);
+        bytes[] memory calldatas = new bytes[](0);
+
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.ProposalInfoArityMismatch.selector));
+        vm.prank(proposer);
+        dao.updateProposal(proposalId, targets, values, signatures, calldatas, '');
+    }
+
+    function test_givenMsgSenderNotProposer_reverts() public {
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.OnlyProposerCanEdit.selector));
+        updateProposal(makeAddr('not proposer'), proposalId, makeAddr('target'), 0, '', '', '');
+    }
+
+    function test_givenPropWithSigners_reverts() public {
+        vm.prank(proposer);
+        dao.cancel(proposalId);
+        (address signer, uint256 signerPK) = makeAddrAndKey('signer');
+        uint256 expirationTimestamp = block.timestamp + 1234;
+        uint256 propId = proposeBySigs(
+            proposer,
+            signer,
+            signerPK,
+            makeTxs(makeAddr('target'), 0, '', ''),
+            'description',
+            expirationTimestamp
+        );
+
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.ProposerCannotUpdateProposalWithSigners.selector));
+        updateProposal(proposer, propId, makeAddr('target'), 1, '', '', 'description');
+    }
+
+    function test_givenStatesPendingActiveSucceededQueuedAndExecuted_reverts() public {
+        // Pending
+        vm.roll(block.number + proposalUpdatablePeriodInBlocks);
+        assertTrue(dao.state(proposalId) == NounsDAOStorageV3.ProposalState.Pending);
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.CanOnlyEditUpdatableProposals.selector));
+        updateProposal(proposer, proposalId, makeAddr('target'), 0, '', '', '');
+
+        // Active
+        vm.roll(block.number + VOTING_DELAY);
+        assertTrue(dao.state(proposalId) == NounsDAOStorageV3.ProposalState.Active);
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOV3Proposals.CanOnlyEditUpdatableProposals.selector));
+        updateProposal(proposer, proposalId, makeAddr('target'), 0, '', '', '');
+
+        // Succeeded
+        vm.prank(proposer);
+        dao.castVote(proposalId, 1);
+        vm.roll(block.number + VOTING_DELAY + VOTING_PERIOD);
```

Comment:
@eladmallel I think here you need to roll only `block.number + VOTING_PERIOD` because you already rolled `VOTING_DELAY` blocks before

Reply:
removed!